Lib/test/test_sys_settrace.py

```diff
@@ -564,294 +576,446 @@ def trace(self, frame, event, arg):
                 self.done = True
         return self.trace
 
-# The first set of 'jump' tests are for things that are allowed:
+# This verifies the line-numbers-must-be-integers rule.
+def no_jump_to_non_integers(output):
+    try:
+        output.append(2)
+    except ValueError as e:
+        output.append('integer' in str(e))
 
-def jump_simple_forwards(output):
-    output.append(1)
-    output.append(2)
-    output.append(3)
+# This verifies that you can't set f_lineno via _getframe or similar
+# trickery.
+def no_jump_without_trace_function():
+    try:
+        previous_frame = sys._getframe().f_back
+        previous_frame.f_lineno = previous_frame.f_lineno
+    except ValueError as e:
+        # This is the exception we wanted; make sure the error message
+        # talks about trace functions.
+        if 'trace' not in str(e):
+            raise
+    else:
+        # Something's wrong - the expected exception wasn't raised.
+        raise RuntimeError("Trace-function-less jump failed to fail")
```

Comment:
`AssertionError` perhaps?

Reply:
I didn't touched this function. Agreed, `AssertionError` looks more appropriate here.